435 multiple shopping lists

lythix opened the pull request:
Have added two new buttons (Save and Import) to the Shopping List panel, and a new button on the Import/Export for importing a shopping list previously saved. 

Save button will prompt for filename and save the blueprints that make up the current shopping list. Import button will ask user if they want to clear current shopping list, or import blueprints without clearing. Seems to work quite nicely, but happy for any feedback from additional testing.

msarilar commented:
excellent, thanks for that !

I'll review this thorougly (I've been away for a while and should come back in around a week)

lythix commented:
> excellent, thanks for that !
> 
> I'll review this thorougly (I've been away for a while and should come back in around a week)

It's been a long week. Any issues with this (I see there are merge conflicts now)

msarilar commented:
hey sorry for the long wait, I had a look back then but didn't get back to it

I wonder if maybe it would be better to use the app's local directory (see `GetManualChangesDirectory`) instead of having the user choose his own, wdyt? that way it stays self contained and EDEngineer doesn't have files/config hanging around

codan-prices commented:
> hey sorry for the long wait, I had a look back then but didn't get back to it

All good, we get busy. I forgot about it too. :)

> 
> I wonder if maybe it would be better to use the app's local directory (see `GetManualChangesDirectory`) instead of having the user choose his own, wdyt? that way it stays self contained and EDEngineer doesn't have files/config hanging around

Sure, sounds good. I'll have a look at making that the default directory and fix up the merge conflicts while I'm at it. Thanks!

msarilar commented:
I fixed the conflicts earlier in the branch

lythix commented:
> I fixed the conflicts earlier in the branch

Have changed the default directory. I've got a new laptop since I last built this branch, and am having some issues with some missing dependencies so can't test it right now. 

I seem to be missing Microsoft.WindowsAPICodePack and possibly some other packages. I'll see if I can work out how I got it building before.

lythix commented on EDEngineer/Views/MainWindow.xaml:
```diff
@@ -198,7 +198,7 @@
                                          VerticalAlignment="Center"
                                          BorderBrush="Transparent"
-                                         FontSize="15"
+                                         FontSize="35"
```
```suggestion
                                         FontSize="15"

```
Not one that I remember. Happy to revert

lythix commented:
Have addressed feedback. 

I'm still having issues building on two different machines. It's been a while since I've done any WPF. 
Not found a fix yet but wondering if you can tell me where your Windows reference path leads to. 
![image](https://user-images.githubusercontent.com/124019/84722729-9b24a480-afb6-11ea-99cb-4d7b202c6a30.png)
As you can see in screenshot, my reference for that dll has an issue finding it which I suspect is my issue building. 

Results in the Windows.UI.Notification using statement not being able to be found
![image](https://user-images.githubusercontent.com/124019/84722776-c1e2db00-afb6-11ea-929f-3de77da27d45.png)

msarilar commented:
thanks very much!

for the Windows.UI.Notifications I think you have to install some UWP SDK or something
